### PR TITLE
federated-graph: new export to generate final API schema from FederatedGraph

### PR DESCRIPTION
**Pull request by tomhoule**
The purpose is for us to stop relying on introspection to generate these schemas. The implementation reuses bits of the existing `render_sdl()` (which is renamed to `render_federated_sdl()` for clarity), but mostly, it is new. In this commit, we modify the composition test setup to reuse the test schemas we already have there.

closes GB-5666

### CODE REVIEW

File: engine/crates/composition/tests/composition/tag_directive_basic/api.graphql [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    HONEYCRISP @tag(name: "appleEnumValue")
+}
+
+type Apple @tag(name: "appleType") {
+    id: ID!
+    texture: Texture
+    variety: AppleVariety
+}
+
+type Orange @tag(name: "orangeType") {
```
Comment (tomhoule) [inline engine/crates/composition/tests/composition/tag_directive_basic/api.graphql:13]: Hmm I think `@tag` shouldn't be in the public API. right?

File: engine/crates/composition/tests/composition/authenticated_basic/api.graphql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+type Car {
```
Comment (Finistere) [inline engine/crates/composition/tests/composition/authenticated_basic/api.graphql:1]: why isn't `@authenticated` part of the api schema  if we show `@deprecated` and `@tag`?

File: engine/crates/composition/tests/composition/requiresScopes_basic/api.graphql [hunk 3]
```diff
@@ -0,0 +1,19 @@
+type B {
```
Comment (Finistere) [inline engine/crates/composition/tests/composition/requiresScopes_basic/api.graphql:1]: I think I would leave `@requiresScopes`

File: engine/crates/composition/tests/composition/requires_basic/api.graphql [hunk 4]
```diff
@@ -0,0 +1,21 @@
+type Farm {
```
Comment (Finistere) [inline engine/crates/composition/tests/composition/requires_basic/api.graphql:1]: I'd consider leaving `@requires`. Contrary to `@provides`, it's not an optimization, it's more related to business logic.

File: engine/crates/composition/tests/composition/tag_directive_basic/api.graphql (hunk 1 above)
Comment (Finistere) [reply engine/crates/composition/tests/composition/tag_directive_basic/api.graphql:13]: No idea for `@tag` to honest. I don't think we should treat it as a public API. The public API is introspection. This is the schema we show to the maintainer/developers of the project. So they need to be aware of the business logic that affects runtime behavior IMHO. We should only strip out the federation aspects (keys, provides, inaccessible, override, etc.)

File: engine/crates/federated-graph/src/render_sdl/render_api_sdl.rs [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+        .any(|directive| matches!(directive, Directive::Inaccessible))
+}
+
+fn write_public_directives(
+    f: &mut fmt::Formatter<'_>,
+    directives: Directives,
+    graph: &FederatedGraphV3,
+) -> fmt::Result {
+    for directive in graph[directives].iter().filter(|directive| match directive {
+        Directive::Authenticated | Directive::Inaccessible | Directive::Policy(_) | Directive::RequiresScopes(_) => {
```
Comment (Finistere) [inline engine/crates/federated-graph/src/render_sdl/render_api_sdl.rs:228]: as mentioned, would include `@authenticated`, `@requiresScopes` and `@requires`. Probably wouldn't bother with `@policy` though, we can't use it...

**Review by Finistere — APPROVED** [review]

File: engine/crates/composition/tests/composition/authenticated_basic/api.graphql (hunk 2 above)
Comment (tomhoule) [reply engine/crates/composition/tests/composition/authenticated_basic/api.graphql:1]: I erred towards removing anything that is federation-specific / is only meant for the engine, not for clients. But there’s really no deep rationale, let’s go with your suggestions.

**Pull request merged**